### PR TITLE
add isCanBeDeselect option

### LINKED ISSUE #64: Adding boolean option to radion button to make selected button can be unselected
So my proposal is, if we can add an boolean option named like `'isRadioCanBeUnselected'`:

- if it true

the selected radio button can be unselected if it tapped again, so the event `onSelected: (index, isSelected)` will return `isSelected = false` if the radio button unselected.
What do you think ?

### PR DESCRIPTION
**Pull request by gnassro**
Fixes #64
I don't know is the name of the option that i added is good or not `isCanBeDeselect`

### CODE REVIEW

File: lib/src/group_button_base.dart [hunk 1]
```diff
@@ -263,12 +266,22 @@ class GroupButton extends StatelessWidget {
...
 
+  /// bool variable for enable radio button to be deselected
+  ///
+  /// * if the [isRadio] = true :
+  /// - if the [isCanBeDeselect] = true , the selected radio button can be deselected
+  /// - if the [isCanBeDeselect] = false , the selected radio button can't be deselected
+  ///
+  /// * if the [isRadio] = false:
+  /// - [isCanBeDeselect] have no effect
+  final bool? isCanBeDeselect;
```
Comment (Frezyx) [inline lib/src/group_button_base.dart:162]: In 4.3.0 version i add GroupButtonOptions model
Can you move this field in the class ?

Comment (Frezyx) [reply lib/src/group_button_base.dart:162]: And you can change name of this field to enableDeselect
I think it's a little clearer

**Review by Frezyx — CHANGES_REQUESTED** [review]

File: lib/src/group_button_body.dart [hunk 2]
```diff
@@ -136,7 +138,7 @@ class _GroupButtonBodyState extends State<GroupButtonBody> {
 
   bool _getSelectedCond(int i) {
     return widget.isRadio
-        ? _controller.selectedIndex == i
+        ? (_controller.selectedIndex ?? i + 1) == i
```
[the remark this answers is not in the document]
Comment (gnassro) [reply lib/src/group_button_body.dart:141]: +1 just to make the condition false, +1, +2, +3... anythings to make the condition false

File: lib/src/group_button_base.dart (hunk 1 above)
Comment (gnassro) [reply lib/src/group_button_base.dart:162]: > In 4.3.0 version i add GroupButtonOptions model Can you move this field in the class ?

but you said that GroupButtonOptions is for the UI settings, did you think this option that i added is can be UI setting ?

File: lib/src/group_button_body.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (Frezyx) [reply lib/src/group_button_body.dart:141]: if selectedIndex == null result will be false in any case

[the remark this answers is not in the document]
Comment (Frezyx) [reply lib/src/group_button_body.dart:141]: It is worth returning this condition to the original version

**Review by Frezyx — APPROVED** [review]
Please return condition to the original view and we will merge the changes

**Review by Frezyx — CHANGES_REQUESTED** [review]
Please return condition to the original view and we will merge the changes

File: lib/src/group_button_body.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (gnassro) [reply lib/src/group_button_body.dart:141]: yes you are right, i will return it

**Review by Frezyx — APPROVED** [review]

**Comment by Frezyx** [comment]
@gnassro 
Thank you for improving the package again!

**Comment by gnassro** [comment]
> @gnassro Thank you for improving the package again!

you're welcome

**Pull request merged**